lib/notifications/display.dart

```diff
@@ -92,7 +93,29 @@ class NotificationDisplayManager {
   static Future<void> _onMessageFcmMessage(MessageFcmMessage data, Map<String, dynamic> dataJson) async {
     assert(debugLog('notif message content: ${data.content}'));
     final zulipLocalizations = GlobalLocalizations.zulipLocalizations;
-    final title = switch (data.recipient) {
+    final groupKey = _groupKey(data);
+    final conversationKey = _conversationKey(data, groupKey);
+
+    final oldMessagingStyle = await ZulipBinding.instance.androidNotificationHost
+      .getActiveNotificationMessagingStyleByTag(conversationKey);
+
+    final MessagingStyle messagingStyle = oldMessagingStyle != null
+      ? MessagingStyle(
+          user: oldMessagingStyle.user,
+          messages: oldMessagingStyle.messages?.toList() ?? [], // Clone a fixed-length list
+          isGroupConversation: oldMessagingStyle.isGroupConversation)
+      : MessagingStyle(
+          user: Person(
+            key: data.userId.toString(),
```

Comment:
Hmm, I wonder what the scope of these Person keys is. Do they apply just within a conversation, or across all the app's notifications?

Unless we find clear confirmation of the former, let's conservatively assume the latter. That means we should include the realm URL in this key, as well as the user ID.

… Ha and I see Chris already asked the same question :smile:: https://github.com/zulip/zulip-flutter/pull/718#discussion_r1663781640

Reply:
I wasn't able to find much info if a Person is scoped to a certain groupKey. So for now, updated it to include realm URL in the person key, just in case.